core/state_transition.go

```diff
@@ -178,24 +179,112 @@ type Message struct {
 	SkipL1Charging bool
 }
 
-type MessageRunMode uint8
+type messageRunMode uint8
 
 const (
-	MessageCommitMode MessageRunMode = iota
-	MessageGasEstimationMode
-	MessageEthcallMode
-	MessageReplayMode
+	messageCommitMode messageRunMode = iota
+	messageGasEstimationMode
+	messageEthcallMode
+	messageReplayMode
 )
 
+type MessageRunContext struct {
+	runMode messageRunMode
+
+	wasmCacheTag uint32
+	wasmTargets  []rawdb.WasmTarget
+}
+
+func NewMessageCommitContext(wasmTargets []rawdb.WasmTarget) *MessageRunContext {
+	if len(wasmTargets) == 0 {
+		wasmTargets = []rawdb.WasmTarget{rawdb.LocalTarget()}
+	}
+	return &MessageRunContext{
+		runMode:      messageCommitMode,
+		wasmCacheTag: 1,
+		wasmTargets:  wasmTargets,
+	}
+}
+
+func NewMessageReplayContext(wasmTargets []rawdb.WasmTarget) *MessageRunContext {
```

Comment:
* Replay and Prefetch shouldn't take targets

* add RecordContext that takes targets

Reply:
only RecordContext and CommitContext should take more targets

Reply:
done, added MessageRecordingContext and removed targets params from Replay and Prefetch contexts constructors